native/spark-expr/src/static_invoke/char_varchar_utils/read_side_padding.rs

```diff
@@ -204,14 +204,21 @@ fn spark_read_side_padding_internal<T: OffsetSizeTrait>(
             );
 
             for (string, length) in string_array.iter().zip(int_pad_array) {
+                let length = length.unwrap();
                 match string {
-                    Some(string) => builder.append_value(add_padding_string(
-                        string.parse().unwrap(),
-                        length.unwrap() as usize,
-                        truncate,
-                        pad_string,
-                        is_left_pad,
-                    )?),
+                    Some(string) => {
+                        if length < 0 {
```

Comment:
its better to put happy path first in `if` stmt for compute intensive parts, so CPU won't have to execute eagerly instructions and then fall it back